rfc-drafts/stream.md

```diff
@@ -0,0 +1,186 @@
+- Feature Name: `async_stream`
+- Start Date: 2020-05-13
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Introduce the `Stream` trait into the standard library, using the
+design from `futures`. Redirect the futures-stream definition to the
+standard library.
+
+# Motivation
+[motivation]: #motivation
+
+XXX describe async streams
```

Comment:
I'd expect to see this motivation section laying out the case specifically for having this trait in the standard library, rather than in futures-core. Some points to include are the perceived increase in stability, the increase in authority, and a promise of support from the language. It may also be worth discussing `stream fn`/`async gen fn` as a goal, as language items like these are one technical reason that would force this trait to be in the standard library (though the timetables for this + the uncertain API for synchronous generators and generator-backed iterators + concerns around "attached"ness make this argument fail to resonate with me at the moment).

Reply:
Yes, this section definitely needs to be elaborated. That's a good list of the points we should be hitting.